app/react-graphql/support/middlewareHelpers.ts

```diff
@@ -22,7 +22,6 @@ function stateFromMiddleWare(
   const combinedState = {
     ...postState,
     variables: {
-      ...preState.variables,
```

Comment:
This was a bigger one that took me a bit.  
We can't merge preState variables here because a middleware will often need to reshape it's variables to match the format expected by graphql.  If we merge on the preState variables it undoes any shaping and causes "unexpected variable" errors.

Reply:
Good catch. I tried this when I was working with bugs combining `usePagination` and `useInfiniteQuery`. I agree that we shouldn't need it.